### PR TITLE
add migration to create new table

### PR DESCRIPTION
**Pull request by jrjamespdx**
## Summary

This change adds the migration for a new database table, CredentialAdoptionEmailRecords.  This table will be used by the Login.gov Adoption Team (under John Rahaghi/OCTO) to store information regarding the on-going Credential Adoption email campaigns while the team works towards implementing Granicus' GovDelivery platform.

Once the implementation of GovDelivery is complete (likely late August/early September), the data contained in this table will be analyzed, appropriate metrics recorded, and then the table will be removed as the GovDelivery platform will handle the data storage needs from there.

## Related issue(s)
This ticket:
https://app.zenhub.com/workspaces/logingov-adoption-team-632280b31e745000136a61fc/issues/gh/department-of-veterans-affairs/va.gov-team/61959

Next ticket once this is merged and deployed: 
https://app.zenhub.com/workspaces/logingov-adoption-team-632280b31e745000136a61fc/issues/gh/department-of-veterans-affairs/va.gov-team/60787

## Testing done

N/A, just a DB migration

## Screenshots
N/A

## What areas of the site does it impact?
Only affects the database for vets-api, no visible components or logic changes associated with this change.

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Review by holytoastr — APPROVED** [review]
Approved per discussion about concerns. Thank you!

File: db/migrate/20230713182204_create_credential_adoption_email_records.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class CreateCredentialAdoptionEmailRecords < ActiveRecord::Migration[6.1]
+  def change
+    create_table :credential_adoption_email_records do |t|
+      t.string :veteran_icn, null: false
+      t.string :veteran_email_address, null: false
```
Comment (rileyanderson) [inline db/migrate/20230713182204_create_credential_adoption_email_records.rb:5]: I'd recommend dropping `veteran_` from these column names. Just to keep it constant with other tables.

Also, this should have an index on `icn` and/or `email` - depending on how you are going to be querying this.

Comment (jrjamespdx) [reply db/migrate/20230713182204_create_credential_adoption_email_records.rb:5]: Hey, @rileyanderson !  

Melissa also brought up the naming convention, so I'll make that change now.

I went back and forth on adding an index on one or both columns.  We anticipate having less than 30,000 records total created for in this database, but we would be checking the table for recent records tied to a particular email in the future so I could see that making use of the query performance gains of having an index.  

If you think the benefit is worth it I can add those indexes in when I do the renaming of the columns here shortly :)

Comment (jrjamespdx) [reply db/migrate/20230713182204_create_credential_adoption_email_records.rb]: changes made!

**Review by moanu404 — APPROVED** [review]

**Review by rmtolmach — APPROVED** [review]
👍 migrations are run automatically.

**Pull request merged**